### PR TITLE
feat: Convert connection-thunk to useConnect hook

### PR DESCRIPTION
**Pull request by sravan-s**
Move logic from sdk/thunks to hooks/useConnect
```
  const reconnect = useConnect({
    appId,
    userId,
    accessToken,
  }, {
    logger,
    nickname,
    profileUrl,
    configureSession,
    customApiHost,
    customWebSocketHost,
    sdk: sdkStore?.sdk,
    sdkDispatcher,
    userDispatcher,
  });
```
Add unit tests
fixes: https://sendbird.atlassian.net/browse/UIKIT-3581

### CODE REVIEW

File: src/lib/hooks/useConnect/connect.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  accessToken,
+  sdk,
+}: ConnectTypes): Promise<void> {
+  await disconnectSdk({
+    logger,
+    sdkDispatcher,
+    userDispatcher,
+    sdk,
+  });
+  await setUpConnection({
```
Comment (AhyoungRyu) [inline src/lib/hooks/useConnect/connect.ts:25]: Should we wrap here \w try - catch just in case disconnectSdk or setUpConnection throws an error?

**Review by HoonBaek — APPROVED** [review]
LGTM

File: src/lib/hooks/useConnect/connect.ts (hunk 1 above)
Comment (sravan-s) [reply src/lib/hooks/useConnect/connect.ts:25]: Before, we used to throw it to customer, can consider wrapping, yeah

Comment (sravan-s) [reply src/lib/hooks/useConnect/connect.ts:25]: Ah, actually wrapped in `useConnect` ~

**Pull request merged**